### DISCUSSION TITLE
Overriding `enabledManagers` in child presets

### DISCUSSION THREAD

**Discussion by sheerlox** [renovatebot/renovate#33321] [Request Help] [answered]
### Tell us more.

Hey :wave: I have a Renovate setup with about 5-6 presets extending in a tree-like structure for each language + project type combinations I'm working on.

I just tried disabling all managers except the ones I need everywhere on my base preset and re-enabling the managers I want in each child preset.

Unfortunately, that doesn't seem possible at the moment without listing every manager in the base preset to disable them.

Would you entertain the idea of letting `{ [manager]: { "enabled": true } }` push to the enabled managers even if it isn't initially included in the `enabledManagers` array?

I'm down to look into this and build a PR if you want :grin: 

Here's the revert commit showing what I was attempting: https://github.com/insurgent-lab/.github/commit/4d4d9377726e7c9f3ddf04a10017225c8928b722

**Comment by rarkins** [comment]
> Would you entertain the idea of letting `{ [manager]: { "enabled": true } }` push to the enabled managers even if it isn't initially included in the `enabledManagers` array?

No, I think that would be too hacky. I recommend trying to use `packageRules` with `matchManagers` + `enabled` settings as these can be overridden more easily.

**Reply by sheerlox** [reply]
Yes, I've ended up using `matchManagers` on all my `packageRules` (see below), it's a bit of work and gives a lot of room to forget one `matchManagers` somewhere, but I guess that's ok. Presets also become unusable (which probably isn't a bad thing in a shared config like this one).

[chore(renovate): refactor to better separate manager configs](https://github.com/insurgent-lab/.github/commit/8e68ce2f9a5bcd144edbe72e88ee8398ac728357)

Not sure I understand what you mean however by `matchManagers` + `enabled`, how do these pair together?

Now that I think of it, you're probably right that it would be hacky to implement such `enabledManagers` overrides. Maybe a new config option like `appendMatchManagers` would make more sense?

**Reply by rarkins** [reply]
You pair together matchManagers + enabled inside packageRules

Can you summarize the logic you want to use?

e.g. by default you want to enable very few managers, but you want to allow repos to enable more, BUT those repos cannot include the original enabledManagers for some reason?

**Reply by sheerlox** [reply]
I'm only trying to achieve that part, the original enabledManagers should remain untouched:

> by default you want to enable very few managers, but you want to allow repos to enable more

Right ok, so I should be able to do something like this:

```js
// main config
{
  matchManagers: ["*"],
  enabled: false,
},
{
  matchManagers: [...DEFAULT_MANAGERS],
  enabled: true,
}
```

```js
// child config
{
  matchManagers: [...SPECIFIC_MANAGERS],
  enabled: true,
}
```

**Reply by rarkins** [reply, marked as answer]
This example you gave should work. The difference between this and enabledManagers is that enabledManagers means that Renovate won't even attempt to scan/discover disabled dependencies, whereas in the above example Renovate will first scan for dependencies and _then_ disable some before looking up

**Reply by sheerlox** [reply]
Just tested this out and it works like a charm. Thank you Rhys!
